From 57de227347eecb45f9fbedb8ad30afe4d24185a2 Mon Sep 17 00:00:00 2001
From: Avula Sri Charan <quic_asrichar@quicinc.com>
Date: Mon, 15 May 2023 18:30:55 +0530
Subject: [PATCH 1/1] ath12k set bw_40 flag in case of EHT40 in 6GHz

Currently for 2GHz & 5Gz we are setting the bw_40 if
IEEE80211_STA_RX_BW_40 is set in ath12k_peer_assoc_h_ht()
where as for 6GHz this case IEEE80211_STA_RX_BW_40 is
not handled in ath12k_peer_assoc_h_he_6ghz because of
which Tx traffic is going in 20MHz when configured in
40MHz.

Fixed it by setting bw_40 flag in case of IEEE80211_STA_RX_BW_40.

Signed-off-by: Avula Sri Charan <quic_asrichar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 84540df..9fa3c72 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -2821,6 +2821,9 @@ static void ath12k_peer_assoc_h_he_6ghz(struct ath12k *ar,
 	if (!arg->he_flag || band != NL80211_BAND_6GHZ || !link_sta->he_6ghz_capa.capa)
 		return;
 
+	if (link_sta->bandwidth == IEEE80211_STA_RX_BW_40)
+		arg->bw_40 = true;
+
 	if (link_sta->bandwidth == IEEE80211_STA_RX_BW_80)
 		arg->bw_80 = true;
 
-- 
2.17.1

